### PR TITLE
Fix bug when shrinking the container in Yarn service

### PR DESCRIPTION
**Pull request by hanghangliu**
update unit test

update requestedContainerCountMap

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1643] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1643

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
There's a bug in "requestTargetNumberOfContainers" method in YarnService when shrinking the container size: it incorrectly updates the requestedContainerCountMap which cause the container count result in wrong number.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Passed unit tests and also tested in cluster.

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java [hunk 1]
```diff
@@ -440,17 +431,12 @@ private EventSubmitter buildEventSubmitter() {
...
     int numTargetContainers = yarnContainerRequestBundle.getTotalContainers();
     // YARN can allocate more than the requested number of containers, compute additional allocations and deallocations
     // based on the max of the requested and actual allocated counts
-    int numAllocatedContainers = this.containerMap.size();
-
-    // The number of allocated containers may be higher than the previously requested amount
-    // and there may be outstanding allocation requests, so the max of both counts is computed here
-    // and used to decide whether to allocate containers.
-    int numContainers = Math.max(numRequestedContainers, numAllocatedContainers);
+    int numAllocatedContainers = Math.max(this.containerMap.size(), allocatedContainerCountMap.values().stream().mapToInt(Integer::intValue).sum());
```
Comment (sv2000) [inline gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java:439]: Why would there be a discrepancy between containerMap.size() and allocatedContainerCountMap size? Maybe add a comment explaining why we are introducing a max() calculation.

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java [hunk 2]
```diff
@@ -462,7 +448,7 @@ public synchronized void requestTargetNumberOfContainers(YarnContainerRequestBun
       for(; requestedContainerCount < desiredContainerCount; requestedContainerCount++) {
         requestContainer(Optional.absent(), yarnContainerRequestBundle.getHelixTagResourceMap().get(currentHelixTag));
       }
-      requestedContainerCountMap.put(currentHelixTag, requestedContainerCount);
+      requestedContainerCountMap.put(currentHelixTag, desiredContainerCount);
```
Comment (sv2000) [inline gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java:451]: I think we discussed this as part of PR#3487: why would desiredContainerCount and requestedContainerCount not be the same here?

**Review by sv2000 — CHANGES_REQUESTED** [review]
A couple of comments seeking clarification.

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java (hunk 1 above)
Comment (hanghangliu) [reply gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java]: Yes, they should be the same. So using containerMap.size() directly

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java (hunk 2 above)
Comment (hanghangliu) [reply gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java:451]: I didn't put this case into consideration: in the desiredContainerCount is smaller than requestedContainerCount, the for-loop wouldn't update the requestedContainerCount. So need to use desiredContainerCount instead.

**Review by sv2000 — APPROVED** [review]
+1. LGTM.

**Pull request merged**